framework/helpers/BaseInflector.php

```diff
@@ -485,6 +485,7 @@ public static function slug($string, $replacement = '-', $lowercase = true)
         }, $parts);
 
         $string = trim(implode($replacement, $replaced), $replacement);
+        $string = preg_replace('#'.preg_quote($replacement).'+#', $replacement, $string);
```

Comment:
> ... s a backwards compatibility break for Yii 2.

add an extra parameter to the method (only for yii2). @samdark ?

```php
public static function slug($string, $replacement = '-', $lowercase = true, $clean = false)
```

I would not implement the extra parameter in yii3.

Reply:
I'd use extra preg_replace or external wrapper or your own implementation of Inflector instead. That's not something that could be considered Yii 2 bug.

Reply:
It was introduced in Yii 2.0.16 and it doesn't look like intended change, so why it's not a bug? In terms of age - 2.0.0 - 2.0.16 is 4 years, 2.0.16 - now is 1.5 year.

Reply:
Hmm... wait, it was doing that before 2.0.16?

Reply:
Alright. That's bug.